src/data/ellpack_page_raw_format.cu

```diff
@@ -1,51 +1,55 @@
-/*!
- * Copyright 2019-2021 XGBoost contributors
+/**
+ * Copyright 2019-2023, XGBoost contributors
  */
-#include <xgboost/data.h>
 #include <dmlc/registry.h>
+#include <xgboost/data.h>
 
+#include <cstddef>  // for size_t
+
+#include "../common/io.h"  // for ResourceReadStream, AlignedFileWriteStream
 #include "ellpack_page.cuh"
-#include "sparse_page_writer.h"
 #include "histogram_cut_format.h"
+#include "sparse_page_writer.h"
 
-namespace xgboost {
-namespace data {
-
+namespace xgboost::data {
 DMLC_REGISTRY_FILE_TAG(ellpack_page_raw_format);
 
 
 class EllpackPageRawFormat : public SparsePageFormat<EllpackPage> {
  public:
-  bool Read(EllpackPage* page, dmlc::SeekStream* fi) override {
+  bool Read(EllpackPage* page, common::ResourceReadStream* fi) override {
     auto* impl = page->Impl();
     if (!ReadHistogramCuts(&impl->Cuts(), fi)) {
       return false;
     }
-    fi->Read(&impl->n_rows);
-    fi->Read(&impl->is_dense);
-    fi->Read(&impl->row_stride);
-    fi->Read(&impl->gidx_buffer.HostVector());
+    if (!fi->Read(&impl->n_rows)) {
```

Comment:
The read method is non-throwing based on the convention set by dmlc core. Therefore we always check the return type and use it to indicate failure. This might not be a good fit for XGBoost.